op_crates/web/08_text_encoding.js

```diff
@@ -138,6 +138,124 @@
     return result;
   }
 
+  function Big5Decoder(big5, bytes, fatal = false, ignoreBOM = false) {
+    if (ignoreBOM) {
+      throw new TypeError("Ignoring the BOM is available only with utf-8.");
+    }
+    const res = [];
+    let lead = 0x00;
+    for (let i = 0; i < bytes.length; i++) {
+      const byte = bytes[i];
+      if (lead !== 0x00) {
+        let pointer = null;
+        const offset = byte < 0x7f ? 0x40 : 0x62;
+        const leadCopy = lead;
+        lead = 0x00;
+        if (inRange(byte, 0x40, 0x7e) || inRange(byte, 0xa1, 0xfe)) {
+          pointer = (leadCopy - 0x81) * 157 + (byte - offset);
+        }
+        if (pointer === 1133) {
+          res.push(202);
+          continue;
+        }
+        if (pointer === 1135) {
+          res.push(202);
+          continue;
+        }
+        if (pointer === 1164) {
+          res.push(234);
+          continue;
+        }
+        if (pointer === 1166) {
+          res.push(234);
+          continue;
+        }
+        const code = pointer === null ? null : big5[pointer];
+        if (code === null && isASCIIByte(byte)) {
+          i--;
+        }
+        if (code === null) {
+          res.push(decoderError(fatal));
+          continue;
+        }
+        res.push(code);
+        continue;
+      }
+      if (isASCIIByte(byte)) {
+        res.push(byte);
+        continue;
+      }
+      if (inRange(byte, 0x81, 0xFE)) {
+        lead = byte;
+        continue;
+      }
+      res.push(decoderError(fatal));
+      continue;
+    }
+    if (lead !== 0x00) {
+      lead = 0x00;
+      res.push(decoderError(fatal));
+    }
+    return res;
+  }
+
+  function Utf16ByteDecoder(
+    bytes,
+    be = false,
+    fatal = false,
+    ignoreBOM = false,
+  ) {
+    if (ignoreBOM) {
+      throw new TypeError("Ignoring the BOM is available only with utf-8.");
+    }
```

Comment:
UTF-16 has BOM too. It is `0xfe, 0xff` / `0xff, 0xfe`. See `wpt/encoding/textdecoder-byte-order-marks.any.js`.